src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseEnvironmentMembersTests.cs

```diff
@@ -0,0 +1,381 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseEnvironmentMembers,
+    Microsoft.NetCore.Analyzers.Runtime.UseEnvironmentMembersFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseEnvironmentMembers,
+    Microsoft.NetCore.Analyzers.Runtime.UseEnvironmentMembersFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseEnvironmentMembersTests
+    {
+        [Fact]
+        public async Task NoDiagnostics_NoEnvironmentProcessId_CSharp()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Diagnostics;
+
+class C
+{
+    void M()
+    {
+        int id = Process.GetCurrentProcess().Id; // assumes Environment.ProcessId doesn't exist
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task NoDiagnostics_NoEnvironmentProcessPath_CSharp()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Diagnostics;
+
+class C
+{
+    void M()
+    {
+        string path = Process.GetCurrentProcess().MainModule.FileName; // assumes Environment.ProcessPath doesn't exist
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task NoDiagnostics_CSharp()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Diagnostics;
+using System.Threading;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static int ProcessId => 0;
+        public static string ProcessPath => """";
+        public static int CurrentManagedThreadId => 0;
+    }
```

Comment:
❔ What reference assemblies do these first appear in?

Reply:
CurrentManagedThreadId:
![image](https://user-images.githubusercontent.com/2642209/109544717-e4118000-7a95-11eb-99df-4afe1ec365c5.png)

ProcessId:
![image](https://user-images.githubusercontent.com/2642209/109544753-ef64ab80-7a95-11eb-81af-1a88ce4036a5.png)

ProcessPath:
Hasn't shipped yet, so .NET 6.